eng/pipelines/pr.yml

```diff
@@ -7,147 +7,540 @@ pr:
-    helixQueueGroup: pr
-    jobParameters:
-      testGroup: innerloop
-      readyToRun: true
-      displayNameArgs: R2R
-
-#
-# CoreFX test runs against CoreCLR
-#
-- template: /eng/platform-matrix.yml
-  parameters:
-    jobTemplate: test-job.yml
-    buildConfig: checked
-    platforms:
-    - Linux_x64
-    - Windows_NT_x64
-    helixQueueGroup: pr
-    jobParameters:
-      testGroup: innerloop
-      corefxTests: true
-      displayNameArgs: CoreFX
-
-#
-# Crossgen-comparison jobs
-#
-- template: /eng/platform-matrix.yml
-  parameters:
-    jobTemplate: crossgen-comparison-job.yml
-    buildConfig: checked
-    platforms:
-    - Linux_arm
-    helixQueueGroup: pr
-
-#
-# Release test builds
-#
-- template: /eng/platform-matrix.yml
-  parameters:
-    jobTemplate: test-job.yml
-    buildConfig: release
-    platforms:
-    - Linux_musl_x64
-    helixQueueGroup: pr
-    jobParameters:
-      testGroup: innerloop
-
-#
-# Formatting
-#
-- template: /eng/platform-matrix.yml
-  parameters:
-    jobTemplate: format-job.yml
-    platforms:
-    - Linux_x64
+stages:
 
+# There are two separate checkout stages: one for Unix-like systems, and one for Windows.
+# With the exception of the archiving format, these stages are mostly the same.  All other
+# build stages depend on these two stages.
+
+- stage: unix_checkout
+  displayName: Checkout (unix)
+  dependsOn: []
+  jobs:
+  - job: checkout
+    pool:
+      ${{ if eq(variables['System.TeamProject'], 'public') }}:
+        vmImage: ubuntu-latest
+      ${{ if ne(variables['System.TeamProject'], 'public') }}:
+        name: NetCoreInternal-Pool
+        queue: BuildPool.Ubuntu.1604.Amd64
+    steps:
+    - checkout: self
+      clean: true
+      fetchDepth: 1
+    - template: /eng/upload-artifact-step.yml
+      parameters:
+        displayName: GIT Repository (unix)
+        rootFolder: $(Build.SourcesDirectory)
+        includeRootFolder: false
+        archiveFile: $(Build.StagingDirectory)/repo_unix.tar.gz
+        archiveType: tar
+        tarCompression: gz
+        artifactName: repo_unix
+
+- stage: windows_checkout
+  displayName: Checkout (windows)
+  dependsOn: []
+  jobs:
+  - job: checkout
+    pool:
+      ${{ if eq(variables['System.TeamProject'], 'public') }}:
+        vmImage: windows-latest
+      ${{ if ne(variables['System.TeamProject'], 'public') }}:
+        name: NetCoreInternal-Pool
+        queue: BuildPool.Windows.10.Amd64.VS2017
+    steps:
+    - checkout: self
+      clean: true
+      fetchDepth: 1
+    - template: /eng/upload-artifact-step.yml
+      parameters:
+        displayName: GIT Repository (windows)
+        rootFolder: $(Build.SourcesDirectory)
+        includeRootFolder: false
+        archiveFile: $(Build.StagingDirectory)/repo_windows.zip
+        archiveType: zip
+        tarCompression: ''
+        artifactName: repo_windows
+
+#
+# Checked builds for Unix-like systems
+#
+
+- stage: build_Linux_arm_checked
+  displayName: Linux arm checked (build)
```

Comment:
I think it is best to remove (build|test) from the individual job names. It duplicates information in the job display.